### ISSUE TITLE
Sample code is pretty terrible

### ISSUE THREAD

**Issue by jamiefraser** [dotnet/docs-maui#591]
Please provide samples that include comments and avoid the use of "myapp" type parameters with no explanation.  Authentication with MSAL now has a pretty bewildering array of flavours - it would be swell to get sample code for a particular platform that takes you through from beginning to end without any mystery variables to figure out

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 829bc017-4a10-6ca5-0717-e1e62f59aacc
* Version Independent ID: 829bc017-4a10-6ca5-0717-e1e62f59aacc
* Content: [Web Authenticator - .NET MAUI](https://docs.microsoft.com/en-us/dotnet/maui/platform-integration/communication/authentication?tabs=ios)
* Content Source: [docs/platform-integration/communication/authentication.md](https://github.com/dotnet/docs-maui/blob/main/docs/platform-integration/communication/authentication.md)
* Product: **dotnet-mobile**
* Technology: **dotnet-maui**
* GitHub Login: @davidbritch
* Microsoft Alias: **dabritch**

**Comment by adegeo** [comment]
@jamiefraser Indeed. I'm unsure where there are examples. There is a basic test available at https://github.com/dotnet/maui/blob/main/src/Essentials/test/DeviceTests/Tests/WebAuthenticator_Tests.cs. There isn't much more to explain with this API, it's extremely simple. You give it a URL to send out to start the authentication, that URL is whatever system for authentication you're using, like talking to Microsoft 365 or something. The URI scheme you give it has been previously registered with the device, which is in the Get Started section. The URI scheme can be anything, like `myapp://` or `awesomeprogram://`.

The diagram does explain the process flow a bit better. It really all depends on the service you're using to authenticate with, you need to learn more about that specific service, like what is required from your app in the URL you send out to.

Perhaps we can explain the `myapp://` a little better? Other than that, I'm not sure there is anything else to explain. What are your thoughts on this?

The diagram in the article does demonstrate the flow.

**Comment by jamiefraser** [comment]
Struggling to find a diagram in this page

https://github.com/dotnet/docs-maui/blob/main/docs/platform-integration/communication/authentication.md

I'm already pretty familiar with MSAL / ADAL and all the concepts. So when I run into snippets like this I find them confusing in the absence of more information:

WebAuthenticatorResult authResult = await WebAuthenticator.Default.AuthenticateAsync(
        new Uri(https://mysite.com/mobileauth/Microsoft),
        new Uri("myapp://"));

    string accessToken = authResult?.AccessToken;

    // Do something with the token
}

The first URI (I'm guessing) is like the Authority endpoint in MSAL, yeah?  Is it relevant when using an AAD endpoint, or only if I'm writing my own auth provider?  What's the "Microsoft" bit at the end?  Is it implying that the endpoint will authenticate Microsoft accounts?

The myapp:// URI looks like it *might* be the redirect URI?  Is that correct?  It's not formatted the way you'd expect if you were using a library like MSAL where the redirect generally includes the App ID.

I stumbled over this article because I'm having issues with the IPublicClientApplication implementation provided for MAUI and thought maybe this article would give some clues as to why the MAUI implementation is broken on iOS as well as Catalyst.  Instead I wound up trying to decode what's relevant and what's throw-away in the snippets provided.  I realize I might sound a bit harsh, but well commented samples are gold.

/j

**Comment by adegeo** [comment]
You'll have to use the published page to see the diagram, I've attached a screenshot at the end of this comment.

Both URIs in the example are purely fictional, so perhaps code comment that is one thing we can do to improve the article. Looking at the diagram, it does say that it's an ASP.NET backend? Perhaps the WebAuthenticator class only works with that type of endpoint? Or is it any endpoint? I'm unsure.

Perhaps @jfversluis can comment on these questions and explain things a little better? Then perhaps we can integrate his comments/thoughts into the article.

> ![image](https://user-images.githubusercontent.com/67293991/172496062-8cedebb2-7908-4cde-818d-35721763d8ee.png)

**Comment by jfversluis** [comment]
To be honest I'm not too familiar with this part of our functionality unfortunately.

**Comment by adegeo** [comment]
@jfversluis do we have a contact that is?

**Comment by davidbritch** [comment]
Hi @jamiefraser 

MSAL support is coming for .NET MAUI, but isn't fully available yet (MSAL requires updates to fully work with .NET MAUI). When it does arrive, there'll be a separate doc on using it.
